src/core/controllers/oidc.go

```diff
@@ -338,3 +434,52 @@ func secretAndToken(tokenBytes []byte) (string, string, error) {
 	}
 	return secret, token, nil
 }
+
+// getSessionType determines if the session is offline by decoding the refresh token or not
+func getSessionType(refreshToken string) (string, error) {
+	parts := strings.Split(refreshToken, ".")
+	if len(parts) != 3 {
+		return "", errors.Errorf("invalid refresh token")
+	}
+	payload, err := base64.RawURLEncoding.DecodeString(parts[1])
+	if err != nil {
+		return "", errors.Errorf("failed to decode refresh token: %v", err)
+	}
+	var claims map[string]interface{}
+	if err := json.Unmarshal(payload, &claims); err != nil {
+		return "", errors.Errorf("failed to unmarshal refresh token: %v", err)
+	}
+	typ, ok := claims["typ"].(string)
+	if !ok {
+		return "", errors.New("missing 'typ' claim in refresh token")
+	}
+	return typ, nil
+}
+
+// revokeOIDCRefreshToken revokes an offline session using the refresh token
+func revokeOIDCRefreshToken(revokeURL, refreshToken, clientID, clientSecret string) error {
+	data := url.Values{}
+	data.Set("token", refreshToken)
+	data.Set("token_type_hint", "refresh_token")
+	auth := base64.StdEncoding.EncodeToString([]byte(clientID + ":" + clientSecret))
+	req, err := http.NewRequest("POST", revokeURL, bytes.NewBufferString(data.Encode()))
+	if err != nil {
+		return errors.Errorf("failed to create request: %v", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Authorization", "Basic "+auth)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
This always skips certificate verification regardless of the oidc_verify_cert setting, creating a security vulnerability.

Reply:
Agree we should respect the setting in OIDC config here since we are interacting with OIDC provider.

I also think this should be moved to "oidc helper" if we think it's general process.

Reply:
resolved.